suite-native/module-accounts-management/src/screens/AccountDetailScreen.tsx

```diff
@@ -19,39 +28,92 @@ import {
 } from '@suite-native/ethereum-tokens';
 import { analytics, EventType } from '@suite-native/analytics';
 import { SettingsSliceRootState } from '@suite-native/module-settings';
+import { BoxSkeleton, Card, VStack } from '@suite-native/atoms';
 
 import { TransactionListHeader } from '../components/TransactionListHeader';
 import { AccountDetailScreenHeader } from '../components/AccountDetailScreenHeader';
 import { TokenAccountDetailScreenSubHeader } from '../components/TokenAccountDetailScreenSubHeader';
 
+const SCREEN_WIDTH = Dimensions.get('window').width;
+
+const cardStyle = prepareNativeStyle(utils => ({
+    padding: utils.spacings.small,
+}));
+
+const LoadingAccountDetailScreen = () => {
+    const { applyStyle } = useNativeStyles();
+
+    return (
+        <Screen screenHeader={<ScreenSubHeader />}>
+            <VStack spacing="extraLarge" alignItems="center">
+                <Card style={applyStyle(cardStyle)}>
+                    <BoxSkeleton width={SCREEN_WIDTH - 32} height={70} />
+                </Card>
+                <Card style={applyStyle(cardStyle)}>
+                    <VStack spacing="large" alignItems="center" paddingHorizontal="large">
+                        <BoxSkeleton width={104} height={104} borderRadius={52} />
+                        <BoxSkeleton width={160} height={30} />
+                        <BoxSkeleton width={200} height={24} />
+                        <BoxSkeleton width={SCREEN_WIDTH - 80} height={48} borderRadius={24} />
+                    </VStack>
+                </Card>
+            </VStack>
+        </Screen>
+    );
+};
+
 export const AccountDetailScreen = memo(() => {
     const route = useRoute<RouteProp<RootStackParamList, RootStackRoutes.AccountDetail>>();
-    const { accountKey, tokenContract } = route.params;
+    const {
+        accountKey: routeAccountKey,
+        tokenContract,
+        networkSymbol,
+        accountType,
```

Comment:
These values cannot be extracted from account that you would get from selector by account key?

Reply:
```
        networkSymbol,
        accountType,
        accountIndex,```
        
These params are needed in case I dont receive accountKey:

User taps add new account and goes directly to account detail before we find/receive accountKey. App uses these 3 to be able to check all accounts whether an account with this symbol, type and index appears and once it happens, app hides loading UI and shows proper detail UI. This is here so we get accountkey from this route and do not have to reload route